### PR TITLE
Refactor GoCardless bank code to avoid duplication

### PR DESCRIPTION
**Pull request by psybers**
The custom bank code typically copies/pastes an existing bank and then modifies it.  So there is a lot of duplicated code.  This is a first pass at refactoring to reuse the fallback code from `integration-bank.js` and avoid duplication.

NOTE: I do not have GoCardless and can not test for breaking changes.  The existing test suite still works but is quite lacking in coverage of these files.

### CODE REVIEW

**Comment by matt-fidd** [comment]
I'm looking at adding quite a lot of test coverage to `actual-server` if that's something that would be welcome?

**Comment by psybers** [comment]
> I'm looking at adding quite a lot of test coverage to `actual-server` if that's something that would be welcome?

More tests are always welcome!

File: src/app-gocardless/banks/american-express-aesudef1.js [hunk 1]
```diff
@@ -30,7 +32,7 @@ export default {
   },
 
   sortTransactions(transactions = []) {
-    return sortByBookingDateOrValueDate(transactions);
+    return Fallback.sortTransactions(transactions);
   },
```
Comment (MatissJanis) [inline src/app-gocardless/banks/american-express-aesudef1.js:36]: **💬 suggestion:** ‏might be easier to remove all these defaults entirely and then just use the object spready syntax. WDYT?

```
export default {
  ...Fallback,
  
   sortTransactions() {
       // custom logic
   },
}
```

Comment (psybers) [reply src/app-gocardless/banks/american-express-aesudef1.js:36]: Could definitely do that.

And then if a new method has to be added down the road, everything just grabs it by default from the Fallback and the banks do not need updated.

Comment (MatissJanis) [reply src/app-gocardless/banks/american-express-aesudef1.js:36]: Yup. That's another benefit (that I definitely thought of before 😄).

**Review by MatissJanis — APPROVED** [review]
Thanks

**Pull request merged**